applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

```diff
@@ -0,0 +1,2556 @@
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+// The member variables related with the MMG library
+MMG5_pMesh mMmgMesh; /// The mesh data from MMG
+MMG5_pSol  mMmgSol;  /// The metric variable for MMG
+// MMG5_pSol  mmgDisp; /// The displacement variable for MMG
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibrary TMMGLibrary>
+void MmgUtilities<TMMGLibrary>::PrintAndGetMmgMeshInfo(
+    MMGMeshInfo<TMMGLibrary>& rMMGMeshInfo,
+    const IndexType EchoLevel
+    )
+{
+    rMMGMeshInfo.NumberOfNodes = mMmgMesh->np;
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        rMMGMeshInfo.NumberOfLines = mMmgMesh->na;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+        rMMGMeshInfo.NumberOfQuadrilaterals = mMmgMesh->nquad;
+    } else { // Surfaces
+        rMMGMeshInfo.NumberOfLines = mMmgMesh->na;
+    }
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        rMMGMeshInfo.NumberOfTetrahedra = mMmgMesh->ne;
+        rMMGMeshInfo.NumberOfPrism = mMmgMesh->nprism;
+    } else { // Surfaces
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+    }
+
+    KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) << "\tNodes created: " << rMMGMeshInfo.NumberOfNodes << std::endl;
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfTriangles + rMMGMeshInfo.NumberOfQuadrilaterals << "\n\tTriangles: " << rMMGMeshInfo.NumberOfTriangles << "\tQuadrilaterals: " << rMMGMeshInfo.NumberOfQuadrilaterals << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTetrahedra + rMMGMeshInfo.NumberOfPrism << "\n\tTetrahedron: " << rMMGMeshInfo.NumberOfTetrahedra << "\tPrisms: " << rMMGMeshInfo.NumberOfPrism << std::endl;
+    } else { // Surfaces
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibrary TMMGLibrary>
+IndexVectorType MmgUtilities<TMMGLibrary>::CheckNodes(
```

Comment:
I still find the function names could be more descriptive.
```suggestion
IndexVectorType MmgUtilities<TMMGLibrary>::FindDuplicateNodeIds(
```

Reply:
Ok